[Shan Hashir] iP

sh4nH opened the pull request:
### Duke Pro

>" Everything begins with an idea" - Earl Nightingale

Duke chat-bot is your pal for keeping track of your to-do list. It is 

- text-based
-  ~EASY~ SUPER EASY to learn and 
- super fast to use

Duke Pro is **free to use** too!
To start using Duke pro today, all you have to do is

1. download it from [here](https://github.com/sh4nH/ip)
2. double click on Duke Pro
3. Add your tasks
4. Let Duke manage your tasks for you 😉

Features of Duke Pro:

- [ ] Manage your tasks (todos,events and deadlines)
- [ ] Reminders (coming soon)
- [ ] Share your tasks with others (coming soon)

____________________________________________________________________________________________________________________________________

If you are interested to learn Java, it can help with that too. Here is the `main` method from DukePro:
```ruby
public static void main(String[] args) {
  new Duke("data/tasks.txt").run();
  }

Commit:
Add Gradle support

sh4nH committed:
Added Greet, Echo, Exit

sh4nH committed:
Added method for user input

sh4nH committed:
Added Add,List

sh4nH committed:
Added Mark as Done

sh4nH committed:
Added ToDos, Events, Deadlines

sh4nH committed:
Added A-TextUiTesting

sh4nH committed:
Handle Errors (empty todo and incorrect statements)

sh4nH committed:
Handel Errors (empty todo and incorrect statements)

sh4nH committed:
Added delete option

sh4nH committed:
Added delete option

sh4nH committed:
Merge branch 'master' of https://github.com/sh4nH/ip

marcuspang commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,12 @@
+    public Deadline(String taskName,String by) {
+        super(taskName);
+        this.by = by;
+    }
```
Your classes and methods require a description as per coding standards. And since you haven't done the `A-JavaDoc` level yet, I won't add a comment for methods/classes missing javadoc.

sh4nH replied:
Hi, I will be adding the description soon. Thank you for reminding me to do it.

marcuspang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String taskName;
```
```suggestion
    protected final String taskName;
```
Since `taskName` is only updated once (i.e. in your constructor), I feel it is good practice to add the `final` keyword. But, I don't think this is part of the coding standards so feel free to close this comment

marcuspang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
```
I like this abstraction for the status icon as it allows for easy modification in the future 👍

marcuspang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+    public  Duke(){
+
+    }
+    ArrayList<Task> tasks = new ArrayList<>();
```
```suggestion
    public Duke() {
    }

    ArrayList<Task> tasks = new ArrayList<>();
```
You have some extra spacing in your constructor function declaration

marcuspang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+
+    }
+    ArrayList<Task> tasks = new ArrayList<>();
+    private final static String horLine = "------------------------------------------------------";
+    private final static String welcomeMsg = horLine + "\nHello! I'm Duke\n" +
```
> Constant names must be all uppercase using underscore to separate words.

Perhaps you can name them something more semantic as well, like `HORIZONTAL_LINE` and `WELCOME_MESSAGE`

sh4nH replied:
Yes, you are correct. I will do that in my next commit.

marcuspang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+        try{
+            //if task is a todo
+            if (taskName.matches("\\btodo\\s.*\\b")){
+                Task newTask = new ToDo(taskName.substring(5));
```
I feel that instead of `newTask` you can name it something else to indicate that it's a `ToDo` class instead of a normal `Task` object (perhaps newToDo). However, this is not part of the coding standards, so feel free to close this

marcuspang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+                System.out.println("\tNow you have " + tasks.size() + " tasks in the list." + "\n" + horLine);
+
+            }
+            ////if task is a deadline
```
```suggestion
            //if task is a deadline
```
I think you might have added additional `/` by accident

sh4nH replied:
oh yes, thank you for pointing it out.

marcuspang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+        } catch(Exception e) {
+            System.out.println(horLine + "\n\t" + e.getMessage() + "\n" + horLine);
+        }
+
+
+
+
+    }
+
+
```
```suggestion
    }

```
>  Logical units within a block should be separated by one blank line.

Although this may technically not be considered a block, these empty lines are unnecessary. There are other instances of this, so I won't comment those to avoid spam

marcuspang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+    private void viewList() {
+        System.out.println(horLine + "\n");
+        System.out.println("\tHere are the tasks in your list:\n");
+        for(int i = 0; i<tasks.size(); i++) {
+            int num = i+1;
```
```suggestion
        for(int i = 0; i < tasks.size(); i++) {
            int num = i + 1;
```
>  White space within a statement

There's quite a lot of instances where whitespace is missing in your statements, which can be solved easily with an IDE. So do a run through your files to make sure they are formatted correctly.

marcuspang reviewed (commented):
There's quite a few coding standard violations, but they can be easily fixed with an IDE. The other issues will be resolved once you reach the `A-JavaDoc` level as well. 

Overall, I think these issues are quite minor, just that there are quite a few of them, and I like the fact that you abstracted the user commands into separate methods. 👍

sh4nH committed:
Added Save feature

sh4nH committed:
Added more OOP

sh4nH committed:
Organized classes into a package

sh4nH committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

sh4nH committed:
Added gradle support

sh4nH committed:
Added JUnit tests

sh4nH committed:
Added Java Docs

sh4nH committed:
Add Coding Standard.

sh4nH committed:
Add find feature.

sh4nH committed:
Add find feature.

sh4nH committed:
Merge branch 'branch-A-JavaDoc'

sh4nH committed:
Merge branch 'branch-A-CodingStandard'

sh4nH committed:
Merge branch 'branch-Level-9'

sh4nH committed:
Add CheckStyle.

sh4nH committed:
Add GUI

sh4nH committed:
Fix main class name in build.gradle

WingKei67 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,128 @@
+
+        userInput.setOnAction((event) -> {
+            handleUserInput();
+        });
```
It would be better if arrow-head style could be avoided.

WingKei67 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+     * @param ui       ui to print the confirmation messages after task is added
+     *                 to the list.
+     */
+    public String addToList(String taskName, Ui ui) {
```
This method may be a bit too long? 🤔

sh4nH committed:
Add Assertions

*Use assert feature (not JUnit assertions).
*Document important assumptions that should hold at various points in the code.

sh4nH committed:
Merge pull request #1 from sh4nH/branch-A-Assertions

Add Assertions

sh4nH committed:
Improve Code Quality
*Critically examine the code and refactor to improve the code quality where necessary.

sh4nH committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/TaskList.java

sh4nH committed:
Add C-Priority

*Provide a way to attach priorities to items(high/low).
*By deafult, a task has low priority.
*To change priority, use 'priority index high/low' command.

sh4nH committed:
Merge pull request #4 from sh4nH/branch-BCD-Extension

Add C-Priority

sh4nH committed:
Update README.md

sh4nH committed:
Update README.md

sh4nH committed:
Update JavaDocs

sh4nH committed:
Handle more exceptions

sh4nH committed:
Update JavaDocs
